src/libraries/System.Text.RegularExpressions/gen/RegexCodeFixer.cs

```diff
@@ -0,0 +1,290 @@
+            if (root is null)
+            {
+                return;
+            }
+
+            SyntaxNode nodeToFix = root.FindNode(context.Span, getInnermostNodeForTie: false);
+
+            if (nodeToFix is null)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    SR.UseRegexSourceGeneratorTitle,
+                    cancellationToken => ConvertToSourceGenerator(context.Document, context.Diagnostics[0], cancellationToken),
+                    equivalenceKey: SR.UseRegexSourceGeneratorTitle),
+                context.Diagnostics);
+        }
+
+        /// <summary>
+        /// Takes a <see cref="Document"/> and a <see cref="Diagnostic"/> and returns a new <see cref="Document"/> with the replaced
+        /// nodes in order to apply the code fix to the diagnostic.
+        /// </summary>
+        /// <param name="document">The original document.</param>
+        /// <param name="diagnostic">The diagnostic to fix.</param>
+        /// <param name="cancellationToken">The cancellation token for the async operation.</param>
+        /// <returns>The new document with the replaced nodes after applying the code fix.</returns>
+        private static async Task<Document> ConvertToSourceGenerator(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            // We first get the compilation object from the document
+            SemanticModel? semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            if (semanticModel is null)
+            {
+                return document;
+            }
+            Compilation compilation = semanticModel.Compilation;
+
+            // We then get the symbols for the Regex and RegexGeneratorAttribute types.
+            INamedTypeSymbol? regexSymbol = compilation.GetTypeByMetadataName(RegexTypeName);
+            INamedTypeSymbol? regexGeneratorAttributeSymbol = compilation.GetTypeByMetadataName(RegexGeneratorTypeName);
+            if (regexSymbol is null || regexGeneratorAttributeSymbol is null)
+            {
+                return document;
+            }
+
+            // Find the node that corresponding to the diagnostic which we will then fix.
+            SyntaxNode? root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (root is null)
+            {
+                return document;
+            }
+
+            SyntaxNode nodeToFix = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: false);
+            // Save the operation object from the nodeToFix before it gets replaced by the new method invocation.
+            // We will later use this operation to get the parameters out and pass them into the RegexGenerator attribute.
+            IOperation? operation = semanticModel.GetOperation(nodeToFix, cancellationToken);
```

Comment:
So seems like you are already computing the operation to fix in the fixer. Wondering if it is easier to avoid the property bag on the diagnostic and just move all the computations to the fixer itself.